### PR TITLE
[WIP] Twig entity embed

### PR DESCRIPTION
**Pull request by stevector**
Here's the corresponding d.o issue. https://drupal.org/node/2279257

It is built on 8.x-1.x right now instead of the embed-display branch. This PR will probably stay Work-In-Progress until it uses  EntityEmbedDisplay.

### CODE REVIEW

**Comment by slashrsm** [comment]
Great job. It also needs some more polishing before we can merge (comments, @todo's, ...)

**Comment by stevector** [comment]
Thanks for the review @slashrsm! I agree I'll need to do more polishing.

File: src/Twig/EntityEmbedTwigExtension.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+   *   The machine name of an view mode like 'default' or 'teaser'.
+   *
+   * @return array
+   *   A render array from entity_view()
+   */
+  public function getRenderArrayByMachineName($entity_id, $entity_type = 'block', $view_mode = 'default') {
+
+    // @todo, use dependency injection instead of procedural code.
+    $entity = entity_load($entity_type, $entity_id);
+    $view = entity_view($entity, $view_mode);
```
Comment (stevector) [inline src/Twig/EntityEmbedTwigExtension.php]: Over in https://github.com/drupal-media/entity_embed/pull/32 I got rid of a direct entity_view() with dependency injection. I will try the same here.

File: src/Twig/EntityEmbedTwigExtension.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+   * @param string $entity_type
+   *   The machine name of an entity_type like 'block'.
+   *
+   * @param string $view_mode
+   *   The machine name of an view mode like 'default' or 'teaser'.
+   *
+   * @return array
+   *   A render array from entity_view()
+   */
+  public function getRenderArrayByMachineName($entity_id, $entity_type = 'block', $view_mode = 'default') {
```
Comment (davereid) [inline src/Twig/EntityEmbedTwigExtension.php]: I wonder if we should tweak these parameters a bit. Put $entity_type (no default) first, then $entity_id_or_uuid, and finally $view_mode = 'default'. Have the second parameter be able to be either a UUID or an entity ID.

Comment (stevector) [inline src/Twig/EntityEmbedTwigExtension.php]: Yeah, you're right. I started this code thinking about blocks first. There isn't much benefit to prioritizing it here. I'll switch it.

File: src/Twig/EntityEmbedTwigExtension.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\entity_embed\Twig\EntityEmbedTwigExtension.
+ */
+
+namespace Drupal\entity_embed\Twig;
+use Drupal\Core\Entity\EntityManagerInterface;
```
Comment (jibran) [inline src/Twig/EntityEmbedTwigExtension.php:10]: Space missing between namespace and use statement.

File: tests/modules/entity_embed_twig_test/templates/entity_embed_twig_test.html.twig [hunk 4]
```diff
@@ -0,0 +1 @@
+{{entity_embed('node', '1') }}
```
Comment (jibran) [inline tests/modules/entity_embed_twig_test/templates/entity_embed_twig_test.html.twig:1]: I'm in love with this line. Thank you @stevector for working on this. And thank you @davereid for encouraging him to work on this.

**Comment by stevector** [comment]
Anyone know how I could go about unit testing getRenderArray() on EntityEmbedTwigExtension?

**Comment by davereid** [comment]
Hrm, that I'm not sure about. I suspect since it involves so much (display plugins, entities themselves, etc), it may be hard to properly unit test.

File: src/Twig/EntityEmbedTwigExtension.php (hunk 2 above)
Comment (davereid) [inline src/Twig/EntityEmbedTwigExtension.php]: :+1:

**Comment by davereid** [comment]
Also, nice work @stevector on incorporating the most recent changes and code available!

**Comment by wimleers** [comment]
Interesting! :)

**Comment by cs-shadow** [comment]
Continuing work on this at #166. Closing this in favor of that PR.

**Pull request closed without merging**